internal/sci/aws/server.go

```diff
@@ -0,0 +1,285 @@
+	}
+
+	subValue := fmt.Sprintf("system:serviceaccount:%s:%s", req.KubernetesNamespace, req.KubernetesServiceAccount)
+
+	// Construct the new trust relationship
+	newTrustRelationship := map[string]interface{}{
+		"Effect": "Allow",
+		"Principal": map[string]interface{}{
+			"Federated": s.OIDCProviderARN,
+		},
+		"Action": "sts:AssumeRoleWithWebIdentity",
+		"Condition": map[string]interface{}{
+			"StringEquals": map[string]string{
+				fmt.Sprintf("%s:sub", s.OIDCProviderURL): subValue,
+			},
+		},
+	}
+
+	// Check if the OIDC provider's trust relationship already exists
+	statements := existingTrustPolicy["Statement"].([]interface{})
+	alreadyExists := false
+	for _, stmt := range statements {
+		stmtMap := stmt.(map[string]interface{})
+		if principal, ok := stmtMap["Principal"].(map[string]interface{}); ok {
+			if federated, ok := principal["Federated"].(string); ok && federated == s.OIDCProviderARN {
+				condition := stmtMap["Condition"].(map[string]interface{})["StringEquals"].(map[string]interface{})
+				condition[fmt.Sprintf("%s:sub", s.OIDCProviderURL)] = subValue
+				alreadyExists = true
+				break
+			}
+		}
+	}
+
+	if !alreadyExists {
+		// Append the new trust relationship to the existing policy
+		existingTrustPolicy["Statement"] = append(statements, newTrustRelationship)
+	}
+
+	updatedTrustPolicy, err := json.Marshal(existingTrustPolicy)
+	if err != nil {
+		return nil, fmt.Errorf("failed to marshal updated trust policy: %v", err)
+	}
+
+	// Apply the updated policy
+	input := &iam.UpdateAssumeRolePolicyInput{
+		PolicyDocument: awsSdk.String(string(updatedTrustPolicy)),
+		RoleName:       awsSdk.String(req.Principal),
+	}
+
+	_, err = s.Clients.IamClient.UpdateAssumeRolePolicy(input)
+	if err != nil {
+		// handle AWS-specific errors
+		// ... (as per your existing error handling code) ...
+		return nil, fmt.Errorf("failed to update trust policy: %v", err)
+	}
+
+	return &sci.BindIdentityResponse{}, nil
+}
+
+func getAccountID(stsSvc *sts.STS) (string, error) {
+	result, err := stsSvc.GetCallerIdentity(&sts.GetCallerIdentityInput{})
+	if err == nil {
+		return *result.Account, nil
+	}
+
+	// Fall back to the environment variable if the sts:GetCallerIdentity call fails
+	envAccountID := os.Getenv("AWS_ACCOUNT_ID")
+	if envAccountID != "" {
+		return envAccountID, nil
+	}
+
+	return "", fmt.Errorf("failed to determine AWS account ID from both STS and environment variable")
+}
+
+func getClusterID() (string, error) {
+	config, err := rest.InClusterConfig()
+	if err != nil {
+		return "", fmt.Errorf("failed to set up in-cluster config: %w", err)
+	}
+
+	clientset, err := kubernetes.NewForConfig(config)
+	if err != nil {
+		return "", fmt.Errorf("failed to create clientset: %w", err)
+	}
+
+	configMapName := "substratus-global"
+	configMapNamespace := "substratus"
+	configMapKey := "cluster_id"
+
+	configMap, err := clientset.CoreV1().ConfigMaps(configMapNamespace).Get(context.TODO(), configMapName, metav1.GetOptions{})
```

Comment:
Prefer mounting ConfigMaps into Pod via env or file over grabbing the ConfigMap. SCI should not need to be assigned Kubernetes RBAC.

Reply:
I think we should ensure the system configmap gets mounted as env variables to sci container as well

Reply:
I did this in my PR just now, you  could cherry-pick this later if my PR isn't merged yet:
https://github.com/substratusai/substratus/pull/185/commits/adeb7832b4c0f28baf67ec10f4ec8d418f40a452

Reply:
I see, yeah that makes good sense. I'll go that path

Reply:
Using env vars now. Its up to the caller to set them. Can do that in follow up or in this PR if we want controller cloud config captured here